subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/testsuites/dependencies/TestSuitesKotlinDSLDependenciesIntegrationTest.groovy

```diff
@@ -0,0 +1,2252 @@
+                    return name;
+                }
+
+                public void setName(String name) {
+                    this.name = name;
+                }
+
+                public int getAge() {
+                    return age;
+                }
+
+                public void setAge(int age) {
+                    this.age = age;
+                }
+            }
+        """
+
+        file('src/test/org/samplePersonTest.java') << """
+            package org.sample;
+
+            import org.apache.commons.beanutils.PropertyUtils;
+
+            public class PersonTest {
+                @Test
+                public void testPerson() {
+                    Object person = new Person();
+                    PropertyUtils.setSimpleProperty(person, "name", "Bart Simpson");
+                    PropertyUtils.setSimpleProperty(person, "age", 38);
+                    assertEquals("Bart Simpson", person.getName());
+                    assertEquals(38, person.getAge());
+                }
+            }
+        """
+
+        expect:
+        succeeds 'checkConfiguration'
+
+        where:
+        desc                | dependencyNotation
+        'GAV string'        | "commons-beanutils:commons-beanutils:1.9.4"
+        'GAV map'           | 'mapOf("group" to "commons-beanutils", "name" to "commons-beanutils", "version" to "1.9.4")'
+    }
+
+    def "can add dependencies using a non-String CharSequence: #type"() {
+        given:
+        buildKotlinFile << """
+        import org.apache.commons.lang3.text.StrBuilder;
+
+        buildscript {
+            ${mavenCentralRepository(GradleDsl.KOTLIN)}
+
+            dependencies {
+                classpath("org.apache.commons:commons-lang3:3.11")
+            }
+        }
+
+        plugins {
+          `java-library`
+        }
+
+        ${mavenCentralRepository(GradleDsl.KOTLIN)}
+
+        val buf: $type = $creationNotation
+
+        testing {
+            suites {
+                val test by getting(JvmTestSuite::class) {
+                    dependencies {
+                        implementation(buf)
+                    }
+                }
+            }
+        }
+
+        tasks.register("checkConfiguration") {
+            dependsOn("test")
+            doLast {
+                val testCompileClasspathFileNames = configurations.getByName("testCompileClasspath").files.map { it.name }
+                assert(testCompileClasspathFileNames.containsAll(listOf("commons-lang3-3.11.jar")))
+            }
+        }
+        """
+
+        expect:
+        succeeds 'checkConfiguration'
+
+        where:
+        type            | creationNotation
+        'StringBuilder' | "StringBuilder(\"org.apache.commons:commons-lang3:3.11\")"
+        'StrBuilder'    | "StrBuilder(\"org.apache.commons:commons-lang3:3.11\")"
+    }
+
+    // region multiple GAV strings
+    def "can NOT add multiple GAV dependencies to #suiteDesc - at the top level (varargs)"() {
```

Comment:
I question the usefulness of this one -- you're actually using the named-args version here, just without the names. The dependency _is_ added, but it isn't resolvable (because it's group and name are full GAV strings).

Reply:
Yep, I suppose it's not clear from the expectations alone that this is expected.  I'll make it clearer this is what's happening.

I think this test is definitely necessary because it demonstrates what happens when you try to use a syntax that works in Groovy, that you would naturally assume works the same way in Kotlin - and it doesn't.  You aren't expecting to invoke the extension method with separate args for each of the GAV coordinates, but that's what ends up happening.  Documenting the limits and differences between the 2 DSLs in one place is the main point of these tests, more important than verifying functionality (which I assume was already done elsewhere).

This is probably another place a contextual error message would be a big help for someone out there who tries this and then wonders what's happening.  The test is at least something to point them towards.

Reply:
I just noticed the `checkConfiguration` method I added here, even though it isn't meant to succeed, contains some assertions that definitely muddy the waters.  I'll clean that up.